### ISSUE TITLE
IBE integration validation request

### ISSUE THREAD

**Issue by adimango** [apaleo/api#283]
## Question

Can you please confirm/verify that we are sending to you all the required details for the creation of a booking request forwarding all the details we get from Adyen.
We followed this docs: https://blog.apaleo.com/developer/ibe-integration-guide

## Background

**apaleo-tracking-id:** 0HLU793HNB0HN:00000050

```
{
   "booker":{
      "email":"*****",
      "phone":"+492312312",
      "address":{
         "city":"22",
         "postalCode":"122",
         "countryCode":"DE",
         "addressLine1":"Hers."
      },
      "lastName":"Musterman",
      "firstName":"Max",
      "couponCode":"",
      "preferedLanguage":null
   },
   "reservations":[
      {
         "adults":1,
         "arrival":"2020-03-26",
         "departure":"2020-03-28",
         "timeSlices":[
            {
               "ratePlanId":"HAR45-MAX1M_NONRE_AP-APARTMENT"
            },
            {
               "ratePlanId":"HAR45-MAX1M_NONRE_AP-APARTMENT"
            }
         ],
         "channelCode":"IBE",
         "externalCode":"be499c4a835916df8c3efb8e34e4b60b-1",
         "corporateCode":null,
         "guaranteeType":"Prepayment",
         "prePaymentAmount":{
            "amount":160,
            "currency":"EUR"
         }
      }
   ],
   "paymentAccount":{
      "isVirtual":false,
      "expiryYear":"2020",
      "payerEmail":"*******",
      "expiryMonth":"10",
      "accountHolder":"Max M",
      "accountNumber":"4305",
      "paymentMethod":"visa",
      "payerReference":"53aeda80-653d-11ea-8464-7b17ec82437c",
      "transactionReference":"883584112466322C"
   }
}
```

**Comment by randyrod** [comment]
Hey @adimango,

I will take a look at it and get back to you.

Randy

**Comment by randyrod** [comment]
In the meantime, we have started moving our developer documentation to [apaleo.dev](https://apaleo.dev)

The direct link to the most updated version of the IBE guide is this: https://apaleo.dev/guides/business-cases/ibe/

**Comment by randyrod** [comment]
Hi @adimango,

We have checked on our logs and see that with that tracking id you got a `200` status code, which means that it went through.

Also, based on the body of the request we can see it is well structured. If there is anything I can help with let me know.

Randy

**Comment by adimango** [comment]
@randyrod thank you for the quick reply.
Can you please confirm we are sending all the details to handle the credit card (paymentAccount) from the Apaleo admin?

**Comment by randyrod** [comment]
Hey @adimango,

From what I can see based on your sample payload, and the result of creating the reservation everything seems to be in order.

Regarding the `deliveryDate` verification, if you are referring to the data that you are sending to Adyen, that information you can get from your Adyen logs, as we do not have that information on our logs.

I will be closing this ticket, but if you have further questions regarding it, you can reopen it and comment on it.

Have a nice day,
Randy

**Comment by adimango** [comment]
Hi @randyrod,
can you please why the following booking request does not update the guest folio.
apaleoTrackingId: 0HLUAHFA3R0GT:000001D5

**Comment by pgrm** [comment]
@adimango this was a GET request. GET requests are always strictly read-only in apaleo

**Comment by adimango** [comment]
What does it mean? It created the reservation

**Comment by pgrm** [comment]
@adimango you asked why the request with that tracking ID doesn't update the guest folio - the tracking ID you provided points to a `GET /booking/v1/reservations?bookingId=IFJFFIZQ` - this is a GET request and doesn't modify any folios.

`IFJFFIZQ` is a booking ID, not a reservation ID. The reservation ID in question would be `IFJFFIZQ-1` could you please rephrase your question, what exactly are you trying to achieve and what exactly is not working?

**Comment by adimango** [comment]
Let me rephrase, the booking request has all the Adyen information but the guest Folio does not update. Do you know why?

**Comment by pgrm** [comment]
ok, @adimango - what do you mean with "the guest folio does not update"? - how should the guest folio update? What is missing on the guest folio?

**Comment by adimango** [comment]
The Guest folio balance stays negative and we get a timeout.

<img width="1359" alt="Screenshot 2020-03-17 at 17 43 13" src="https://user-images.githubusercontent.com/5462772/76879671-dd82fc00-6876-11ea-8d68-84732f21f1d2.png">

**Comment by pgrm** [comment]
@adimango yes, there is a configuration from your old account which you didn't update properly.

You're providing `"metadata.accountId":"YOUR_TEST_ACCOUNT_ID"` there you should provide the production account ID, otherwise apaleo payment can't correlate the authorization request together with the capture request

**Comment by adimango** [comment]
Got it, we forgot to update the ENV variable. thank you @pgrm

**Comment by bennetone** [comment]
@adimango Ideally you do not configure your account id, but just call `GET https://api.apaleo.com/account/v1/accounts/current` to load the account id of your current environment. Less configuration means usually less problems.
